azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -937,6 +958,94 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
 }
 
+func (KubernetesClusterNodePoolResource) kubeletAndLinuxOSConfig(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "test" {
+  name                  = "internal"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = "Standard_DS2_v2"
+  node_count            = 1
+
+  kubelet_config {
+    cpu_manager_policy        = "static"
+    cpu_cfs_quota_enabled     = true
+    cpu_cfs_quota_period      = "10ms"
+    image_gc_high_threshold   = 90
+    image_gc_low_threshold    = 70
+    topology_manager_policy   = "best-effort"
+    allowed_unsafe_sysctls    = ["kernel.msg*", "net.core.somaxconn"]
+    container_log_max_size_mb = 100
+    container_log_max_line    = 100000
+    pod_max_pid               = 12345
```

Comment:
since all of these are Optional - we should have a test setting only some of these fields

Reply:
done